puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/foreman-proxy-develop-test.groovy

```diff
@@ -0,0 +1,89 @@
+def commit_hash = ''
+
+pipeline {
+    agent any
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage("Collect Git Hash") {
+            steps {
+                git(url: 'https://github.com/theforeman/smart-proxy', branch: 'develop')
+                script {
+                    commit_hash = archive_git_hash()
+                }
+            }
+        }
+        stage("test ruby-2.1 & puppet-4.10.9") {
+            steps {
+                run_test(ruby: '2.1', puppet: '4.10.9')
+            }
+        }
+        stage("test ruby-2.2 & puppet-4.10.9") {
+            steps {
+                run_test(ruby: '2.2', puppet: '4.10.9')
+            }
+        }
+        stage("test ruby-2.3 & puppet-4.10.9") {
+            steps {
+                run_test(ruby: '2.3', puppet: '4.10.9')
+            }
+        }
+        stage("test ruby-2.4 & puppet-4.10.9") {
+            steps {
+                run_test(ruby: '2.4', puppet: '4.10.9')
+            }
+        }
+        stage("test ruby-2.4 & puppet-5.3.3") {
+            steps {
+                run_test(ruby: '2.4', puppet: '5.3.3')
+            }
+        }
+        stage("test ruby-2.5 & puppet-4.10.9") {
+            steps {
+                run_test(ruby: '2.5', puppet: '4.10.9')
+            }
+        }
+        stage("test ruby-2.5 & puppet-5.3.3") {
+            steps {
+                run_test(ruby: '2.5', puppet: '5.3.3')
+            }
+        }
+        stage("Release Nightly Package") {
+            steps {
+                build(
```

Comment:
This will fail the job if the RPM or Debian don't build IIRC. I've always debated internally if that was the behavior we wanted. Given this is the test job that produces a tested and valid commit hash. My thinking is we don't want this job to be marked as failed just because the release failed. I think that would mean either marking this as don't fail if this build fails or to reverse the triggering (i.e. put the trigger on the release job).